File: data/src/main/java/com/ykb/app/cryptotrader/data/dao/UserDao.java
package com.ykb.app.cryptotrader.data.dao;

import com.ykb.app.cryptotrader.data.model.User;
import com.ykb.app.cryptotrader.data.repo.UserRepo;
import org.springframework.stereotype.Component;

@Component
public final class UserDao extends Dao<User, Long> {

    public UserDao(UserRepo repo) {
        super(repo);
    }

    public boolean usernameExists(String username) {
        return ((UserRepo) repo).usernameExists(username);
    }

}

File: library/data/data-auth/src/main/kotlin/com/ykb/app/cryptotrader/data/auth/dao/UserDao.kt
package com.ykb.app.cryptotrader.data.auth.dao

import com.ykb.app.cryptotrader.data.auth.model.User
import com.ykb.app.cryptotrader.data.auth.repo.UserRepo
import com.ykb.app.cryptotrader.data.base.Dao
import org.springframework.stereotype.Service

@Service
class UserDao(repo: UserRepo) : Dao<User, Int>(repo)